apis/kubedb/v1alpha2/mysql_types.go

```diff
@@ -180,6 +181,29 @@ type MySQLTopology struct {
 	// and it will take reference of  appbinding of the source
 	// +optional
 	ReadReplica *MySQLReadReplicaSpec `json:"readReplica,omitempty"`
+	// +optional
+	SemiSync *SemiSyncSpec `json:"semiSync,omitempty"`
+}
+
+// +kubebuilder:validation:Enum= Clone;PseudoTransaction
+
+type ErrantTransactionRecoveryPolicy string
+
+const (
+	ErrantTransactionRecoveryPolicyClone             ErrantTransactionRecoveryPolicy = "Clone"
+	ErrantTransactionRecoveryPolicyPseudoTransaction ErrantTransactionRecoveryPolicy = "PseudoTransaction"
+)
+
+type SemiSyncSpec struct {
+	// count of slave to wait for before commit
+	// +kubebuilder:default=1
+	//+kubebuilder:valida+kubebuilder:validation:Minimum=1
+	SourceWaitFoReplicaCount int `json:"SourceWaitFoReplicaCount,omitempty"`
```

Comment:
`json:"SourceWaitFoReplicaCount

should be small `s` `sourceWaitFoReplicaCount`